### PR TITLE
don't use setAccessible on java 9

### PR DESCRIPTION
**Pull request by ndeloof**
Signed-off-by: Nicolas De Loof <nicolas.deloof@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,7 +99,7 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>animal-sniffer-maven-plugin</artifactId>
-        <version>1.15</version>
+        <version>1.17-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml]: You must deploy a timestamped snapshot for CI to proceed. FTR my `~/.bashrc`:

```bash
alias deploysnapshot='mvn clean install source:jar deploy:deploy -DaltDeploymentRepository=maven.jenkins-ci.org::default::https://repo.jenkins-ci.org/snapshots/ -DskipTests'
```

Also please include a `TODO` comment in the POM linking to the upstream PR so we can see when it is merged and released.

File: core/pom.xml [hunk 2]
```diff
@@ -25,6 +25,11 @@
     </plugins>
   </build>
   <dependencies>
+    <dependency>
+      <groupId>io.jenkins.fields</groupId>
+      <artifactId>fields</artifactId>
+      <version>1.0.0-20180620.094913-3</version>
```
Comment (jglick) [inline core/pom.xml]: Please link to upstream PR, or otherwise give some indication in the POM of how we would know when this is ready to bump to a release version.

File: core/src/main/java/org/kohsuke/stapler/ResponseImpl.java [hunk 3]
```diff
@@ -344,7 +344,7 @@ public int reverseProxyTo(URL url, StaplerRequest req) throws IOException {
 
         // copy the response
         int code = con.getResponseCode();
-        setStatus(code,con.getResponseMessage());
+        sendError(code,con.getResponseMessage());
```
Comment (jglick) [inline core/src/main/java/org/kohsuke/stapler/ResponseImpl.java]: How is this related?

File: core/src/main/java/org/kohsuke/stapler/jsr269/QueryParameterAnnotationProcessor.java [hunk 4]
```diff
@@ -76,7 +76,7 @@ private void write(ExecutableElement m) throws IOException {
 
         OutputStream os = f.openOutputStream();
         try {
-            IOUtils.write(buf, os, "UTF-8");
+            IOUtils.write(buf.toString(), os, "UTF-8");
```
Comment (jglick) [inline core/src/main/java/org/kohsuke/stapler/jsr269/QueryParameterAnnotationProcessor.java]: How is this related?

**Comment by ndeloof** [comment]
this PR was for demonstration purpose during Java 10 hackathon, not relevant anymore
